src/lib/validationRules/staticClientIdUri/staticClientIDUri.ts

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright 2022 Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import { ValidationRule, ValidationContext } from "../../types";
+
+const staticClientIdUri: ValidationRule = {
+  rule: {
+    type: "local",
+    name: "Client Identifier URI should not have search parameters attached",
+    description:
+      "The Client Identifier Document should be hosted statically. Therefore, search parameters hint at an anti-pattern.",
```

Comment:
That's not necessarily true. Why would this be an issue?

Reply:
Mh, this is something @ThisIsMissEm advised me to do. I suppose for the reason that a Client Identifier Document should not change over time and a dynamic URI might hint at an anti-pattern here?
Should I leave it out?

Reply:
I'd leave it out for the time being. We want to make minimal assumptions about what people are doing, and nothing prevents from serving a client identifier from a dynamic IRI, or to use the query parameter for routing. I'd rather add rules as we observe issue rather than preemptively preventing behaviours that may not be the most common use case, but are valid and do not create interop issues.